draft-ietf-lamps-csr-attestation.md

```diff
@@ -786,78 +786,257 @@ The second example conveys an Arm Platform Security Architecture token,
+~~~
+where for a key object the attested field is
+
+~~~
+     typedef struct {
+          TPM2B_NAME name; 
+          TPM2B_NAME qualifiedName;
+     } TPMS_CERTIFY_INFO;
+~~~
+
+#### TPM2 Public Structure
+
+Any TPM2 Object has an associated TPM2 Public structure defined
+as TPMT_PUBLIC. This is defined below as a 'C' structure. While there
+are many types of TPM2 Objects each with its own specific TPMT_PUBLIC
+structure (handled by the use of 'unions') this document will specifically
+define TPMT_PUBLIC for a TPM2 key object.
+
+~~~
+     typedef struct {
+          TPMI_ALG_PUBLIC type; 
+          TPMI_ALG_HASH nameAlg; 
+          TPMA_OBJECT objectAttributes; 
+          TPM2B_DIGEST authPolicy; 
+          TPMU_PUBLIC_PARMS parameters; 
+          TPMU_PUBLIC_ID unique;
+     } TPMT_PUBLIC; 
+~~~
+
+Where:
+* type and nameAlg are 16 bit TCG defined algorithms.
+* objectAttributes is a 32 bit field defining properties of the object, as shown below
+
+~~~
+     typedef struct TPMA_OBJECT {
+          unsigned Reserved_bit_at_0 : 1; 
+          unsigned fixedTPM : 1; 
+          unsigned stClear : 1; 
+          unsigned Reserved_bit_at_3 : 1; 
+          unsigned fixedParent : 1; 
+          unsigned sensitiveDataOrigin : 1; 
+          unsigned userWithAuth : 1; 
+          unsigned adminWithPolicy : 1; 
+          unsigned Reserved_bits_at_8 : 2; 
+          unsigned noDA : 1; 
+          unsigned encryptedDuplication : 1; 
+          unsigned Reserved_bits_at_12 : 4; 
+          unsigned restricted : 1; 
+          unsigned decrypt : 1; 
+          unsigned sign : 1; 
+          unsigned x509sign : 1; 
+          unsigned Reserved_bits_at_20 : 12; 
+     } TPMA_OBJECT; 
+~~~
+
+* authPolicy is the Policy Digest needed to authorize use of the object.
+* Parameters are the object type specific public information about the key.
+     * For key objects, this would be the key's public parameters. 
+* unique is the identifier for parameters
+
+The size of the TPMT_PUBLIC is provided by the following structure:
+
+~~~
+     typedef struct {
+          UINT16     size;
+          TPMT_PUBLIC publicArea;
+     } TPM2B_PUBLIC;
+~~~
+
+#### TPM2 Signatures
+
+TPM2 signatures use a union where the first field (16 bits) identifies
+the signature scheme. The example below shows an RSA signature where
+TPMT_SIGNATURE->sigAlg will indicate to use TPMS_SIGNATURE_RSA
+as the signature.
+
+~~~
+     typedef struct {
+          TPMI_ALG_SIG_SCHEME sigAlg; 
+          TPMU_SIGNATURE signature; 
+     } TPMT_SIGNATURE;
+
+     typedef struct {
+          TPMI_ALG_HASH hash; 
+          TPM2B_PUBLIC_KEY_RSA sig; 
+     } TPMS_SIGNATURE_RSA;
+~~~
+
+#### Attestation Key {#attestation-key}
+
+The uniquely identifying TPM2 key is the Endorsement Key (the EK). As this is a privacy
+sensitive key, the EK is not directly used to attest to any TPM2 asset. Instead,
+the EK is used by an Attestation CA to create an Attestation Key (the AK). The AK is
```

Comment:
Hello! 👋🏻 Great write-up! I was reading through and got a bit confused by this line, specifically by `the EK is used by an Attestation CA to create an Attestation Key`. My understanding was that the EK is used to assure a 3rd party (in this case the Privacy CA / Attestation CA) that the AK is in the same TPM as the EK, not for the CA to issue the AK. Am I getting this wrong?

Reply:
I think we are saying the same thing. The Attestation CA is just a special CA that uses the TPM protocols (including the EK) to prove the AK is in the same TPM (and has the expected attributes). The Attestation CA then issues a certificate for the AK. Since the AK is a signing key (the EK is a decryption key so the protocol is a bit awkward and non-standard - again, this is one of the reasons we use the term Attestation CA because it acts as a non-standard CA) it is easier to get a signed statement about the properties of a "user key". Defining that signed statement is the point of this effort.

Reply:
Agree, but I don't see how that maps to `used by an Attestation CA to create an Attestation Key` - perhaps `used by an Attestation CA to create an Attestation Key certificate` is clearer?

Reply:
Ha, you are correct. Authors sometimes miss the subtitle wording and when I read ...create and Attestation Key... I meant what you said. A subtle but very important difference. Thanks for pointing this out.